java/code/src/com/redhat/rhn/domain/server/Server.java

```diff
@@ -2194,6 +2201,11 @@ public String getChannelHost() {
                 .orElseGet(() -> ConfigDefaults.get().getCobblerHost());
     }
 
+    public String getHubServerHost() {
+        return this.getLastServerPath().map(p -> p.getHostname())
+                .orElseGet(() -> ConfigDefaults.get().getCobblerHost());
+    }
```

Comment:
I think this is not correct. What happens when we do not have a Hub scenario?
I think we call getHubServerHost() and we get an answer. But this is just a server and not a Hub. Is this wanted?

And the javadoc is missing

Reply:
Javadoc added.

Reply:
The method returns the Hub server (last in the chain) or the server itself if no proxy is present.